### PR TITLE
implicit-0.0.3 fails to build on ghc-7.10.1: No instance for (Applicative ArgParser)

### PR DESCRIPTION
**Pull request by l29ah**
Graphics/Implicit/ExtOpenScad/Util/ArgParser.hs:13:10:
    No instance for (Applicative ArgParser)
      arising from the superclasses of an instance declaration
    In the instance declaration for ‘Monad ArgParser’

Graphics/Implicit/ExtOpenScad/Util/ArgParser.hs:31:10:
    No instance for (GHC.Base.Alternative ArgParser)
      arising from the superclasses of an instance declaration
    In the instance declaration for ‘MonadPlus ArgParser’

### CODE REVIEW

File: implicit.cabal [hunk 1]
```diff
@@ -43,6 +43,7 @@ Library
         -rtsopts
         -funfolding-use-threshold=16 
         -fspec-constr-count=10
+        -fno-warn-tabs
```
Comment (lancelet) [inline implicit.cabal]: Drive-by review: Has `-fno-warn-tabs` been added because there are tabs in the current code?

It might be useful to remove them as well, as part of a 7.10 patch.

Comment (l29ah) [inline implicit.cabal]: On Sun, Jun 28, 2015 at 05:19:52PM -0700, Jonathan Merritt wrote:
> Drive-by review: Has `-fno-warn-tabs` been added because there are tabs in the current code?

Yep.

> It might be useful to remove them as well, as part of a 7.10 patch.

I don't see tabs as a bad thing anyway.

-- 
()  ascii ribbon campaign - against html mail
/\  http://arc.pasp.de/   - against proprietary attachments

Comment (lancelet) [inline implicit.cabal]: Lol... since 7.10, ghc disagrees with you on tabs, but I'm not in a position to. :smile:

Nice work on the rest, getting it up to speed with the AMP changes.
